content/127labs/drone-k8s-helm/index.md

```diff
@@ -0,0 +1,54 @@
+---
+date: 2017-06-10T00:00:00+00:00
+title: Kubernetes Helm
+author: 127labs
+tags: [ kubernetes, helm, docker ]
+repo: 127labs/drone-k8s-helm
+logo: kubernetes_helm.svg
+image: 127labs/drone-k8s-helm
+---
+
+This Kubernetes plugin can be included into your pipelines to allow deployment with Helm.
+
+Example configuration:
+
+```yaml
+pipeline:
+  ship:
+    image: 127labs/drone-k8s-helm
+    chart: stable/dokuwiki
+    master: https://127.0.0.1:8001
+    token: super-long-token
+    release: wiki
+    skip_tls: true
+    clean_before_release: true
+    values:
+      dokuwikiEmail: $${DOKUWIKI_EMAIL}
+      dokuwikiPassword: $${DOKUWIKI_PASSWORD}
+    secrets: [dokuwiki_email, dokuwiki_password]
+```
+
+# Secret Reference
+
+Values are capable of accessing secrets by interpolating them with the `$${SECRET}` syntax. The plugin will
```

Comment:
thanks for the reply!

One of the things that jumped out is the use of `${VARIABLE}`. In older versions of drone we injected secrets into the yaml using this exact syntax. We also use this syntax to inject build metadata into the yaml (such as branch, sha, etc). So there is a bit of history with this syntaxthat could confuse people.

One option could be separating hard-coded variables and variables sourced from the environment. The latter would map environment variables to use-defined names.

```diff
values:
  dokuwikiEmail: 'hardcodedEmail'
  dokuwikiPassword: 'hardcodedPassword'
+values_from_env:
+  dokuwikiEmail: DOKUWIKI_EMAIL
+  dokuwikiPassword: DOKUWIKI_PASSWORD
```

So for the record, nothing wrong with the implementation. More just the history of that specific syntax that tripped me up initially :)